src/allmydata/storage_client.py

```diff
@@ -126,6 +128,27 @@ def _check_connected_high_water_mark(self):
                 remaining.append( (threshold, d) )
         self._threshold_listeners = remaining
 
+    def _ensure_tub_created(self, serverid):
+        if serverid in self.tubs:
+            return
+        self.tubs[serverid] = Tub()
+
+        # XXX set options?
+        self.tubs[serverid].setServiceParent(self)
+
+    def got_static_announcement(self, key_s, ann):
+        if key_s is not None:
```

Comment:
`key_s` should always be present now (since we got rid of v1 introducers). Let's remove the `if` and do these precondition checks all the time. Let's also make the `assert` on the service-name into a `precondition()`.